gnosis/safe-react
# Add multisend contract details and delegate call warning

## What it solves
Resolves #3090

## How this PR fixes it
- An `EthHashInfo` component has been added to Multisend transaction details, displaying the details off the contract being interacted with.
- All delegate calls now verify whether a name was returned by the CGW and shows a warning when none was (an unexpected delegate call).

## How to test it
1. If a delegate call has a known address, "Delegate Call" will be shown as before.
2. If a delegate call is not a known address. "⚠️ Unexpected Delegate Call" in red.

- Create a multisend transaction and verify that the multisend address is now shown.
  - On Rinkeby, we do not have the latest Multisend contract version as a known address. It should show as unknown.
  - On other networks, it will be a normal delegate call.

## Screenshots

[Multisend address shown](https://pr3093--safereact.review-safe.gnosisdev.com/app/rin:0x501E66bF7a8F742FA40509588eE751e93fA354Df/transactions/multisig_0x501E66bF7a8F742FA40509588eE751e93fA354Df_0x2554666e681aa77d25b2f53ee76b32bfa409aa9e69916cde8a7df38cab886fd2) (delegate call):
![image](https://user-images.githubusercontent.com/20442784/144449561-17456bf9-a7e7-4c1a-93ca-aeb45e29462d.png)

[Multisend address shown](https://pr3093--safereact.review-safe.gnosisdev.com/app/rin:0x501E66bF7a8F742FA40509588eE751e93fA354Df/transactions/multisig_0x501E66bF7a8F742FA40509588eE751e93fA354Df_0x1e25374e28df19ca4911c2bf2d674a458414c0bef9a151284dbfa3dea11752d6) (unexpected delegate call):
![image](https://user-images.githubusercontent.com/20442784/144449468-8abf3789-57ff-46ee-8262-74cfa0d51e12.png)

[Delegate call](https://pr3093--safereact.review-safe.gnosisdev.com/app/eth:0x1f7f8DA3eac80DBc0b4A49BC447A88585D8766C8/transactions/multisig_0x1f7f8DA3eac80DBc0b4A49BC447A88585D8766C8_0xce8a0adcdefe81baf4c33222779ca9b406b89e472e0f3942791a81bc91c93163), as before:
![image](https://user-images.githubusercontent.com/20442784/144415463-476214b9-dd51-48bf-ab96-0773804e1525.png)

[Unexpected delegate call warning](https://pr3093--safereact.review-safe.gnosisdev.com/app/eth:0x1f7f8DA3eac80DBc0b4A49BC447A88585D8766C8/transactions/multisig_0x89a3C5C9880Efc029984de67b81aba0FC45b44d8_0xd1c0929e911036b1f09bb819e603bb69d2651e4f553dc0d49479c5e280d7d6cd):
![image](https://user-images.githubusercontent.com/20442784/144415387-78f8a408-be3a-4286-a38d-95fc6bb349ec.png)
